packages/typespec-autorest-canonical/src/lib.ts

```diff
@@ -0,0 +1,273 @@
+   * This extension is meant for debugging and should not be depended on.
+   * @default "never"
+   */
+  "include-x-typespec-name"?: "inline-only" | "never";
+
+  /**
+   * Path to the common-types.json file folder.
+   * @default "${project-root}/../../common-types/resource-management"
+   */
+  "arm-types-dir"?: string;
+
+  /**
+   * Determines whether to transmit the 'readOnly' property to lro status schemas.
+   * @default false
+   */
+  "use-read-only-status-schema"?: boolean;
+}
+
+const EmitterOptionsSchema: JSONSchemaType<AutorestCanonicalEmitterOptions> = {
+  type: "object",
+  additionalProperties: false,
+  properties: {
+    "output-file": {
+      type: "string",
+      nullable: true,
+      description: [
+        "Name of the output file.",
+        "Output file will interpolate the following values:",
+        " - service-name: Name of the service if multiple",
+        " - azure-resource-provider-folder: Value of the azure-resource-provider-folder option",
+        "",
+        "Default: `{azure-resource-provider-folder}/{service-name}/{version}/openapi.json`",
+        "",
+        "",
+        "Example: Single service",
+        " - `canonical.openapi.json`",
+        "",
+        "Example: Multiple services",
+        " - `Service1.canonical.openapi.json`",
+        " - `Service2.canonical.openapi.json`",
+      ].join("\n"),
+    },
+    version: { type: "string", nullable: true },
+    "azure-resource-provider-folder": { type: "string", nullable: true },
+    "arm-types-dir": {
+      type: "string",
+      nullable: true,
+      description:
+        "Path to the common-types.json file folder. Default: '${project-root}/../../common-types/resource-management'",
+    },
+    "new-line": {
+      type: "string",
+      enum: ["crlf", "lf"],
+      nullable: true,
+      default: "lf",
+      description: "Set the newline character for emitting files.",
+    },
+    "omit-unreachable-types": {
+      type: "boolean",
+      nullable: true,
+      description:
+        "Omit unreachable types. By default all types declared under the service namespace will be included. With this flag on only types references in an operation will be emitted.",
+    },
+    "include-x-typespec-name": {
+      type: "string",
+      enum: ["inline-only", "never"],
+      nullable: true,
+      default: "never",
+      description:
+        "If the generated openapi types should have the `x-typespec-name` extension set with the name of the TypeSpec type that created it.\nThis extension is meant for debugging and should not be depended on.",
+    },
+    "use-read-only-status-schema": {
+      type: "boolean",
+      nullable: true,
+      default: false,
+      description: "Create read-only property schema for lro status",
+    },
+  },
+  required: [],
+};
+
+const libDef = {
+  name: "@azure-tools/typespec-autorest-canonical",
+  diagnostics: {
+    "security-service-namespace": {
```

Comment:
Can you go through all those diagnostics and make sure you emit them, I know this one is not used in autorest anymore so probably not in your emitter either.

Reply:
you can just delete them one by one and check if the `watch` build produce any error, if it does it means it is used

Reply:
Made this pr removing the autorest ones not used https://github.com/Azure/typespec-azure/pull/408

Reply:
updated